system-addon/lib/TopSitesFeed.jsm

```diff
@@ -161,6 +161,9 @@ this.TopSitesFeed = class TopSitesFeed {
 
         // Remove internal properties that might be updated after dispatch
         delete link.__sharedCache;
+
+        // Indicate that these links should get a frecency bonus when clicked
+        link.typedBonus = true;
```

Comment:
We could and it would probably have fewer changes now, but then that would mean checking the type/source in various places. Similar to the issue with the recent download type, there would need to be special casing to allow for specific behaviors, and if other things want that behavior too, we probably just end up having a larger condition instead of the single check for the actual desired behavior.

Also, conceptually, should the feed be the one controlling the frecency behavior or something that the UI happens to control?

Reply:
>the feed be the one controlling the frecency behavior 

:+1: